File: bookstore-frontend/src/services/bookService.js
import api from './api';

const bookService = {
  getAllBooks: () => api.get('/books'),
  getBookById: (id) => api.get(`/books/${id}`),
  createBook: (bookData) => api.post('/books', bookData),
  updateBook: (id, bookData) => api.put(`/books/${id}`, bookData),
  deleteBook: (id) => api.delete(`/books/${id}`),
  searchBooks: (query) => api.get(`/books/search?query=${query}`),
};

export default bookService;

File: src/main/java/com/bookstore/booksmanagementsystem/service/BookService.java
package com.bookstore.booksmanagementsystem.service;

import com.bookstore.booksmanagementsystem.dto.BookDTO;

import java.util.List;

public interface BookService {
	BookDTO createBook(BookDTO bookDTO);

	BookDTO getBookById(Long id);

	List<BookDTO> getAllBooks();

	BookDTO updateBook(Long id, BookDTO bookDTO);

	void deleteBook(Long id);

	List<BookDTO> searchBooks(String query);

	BookDTO updateBookImage(Long id, String imageUrl);
}
